probe-rs/src/bin/probe-rs/cmd/test.rs

```diff
@@ -0,0 +1,617 @@
+use std::cell::RefCell;
+use std::fs::File;
+use std::io::Write;
+use std::ops::Range;
+use std::path::Path;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use anyhow::{anyhow, bail, Context, Result};
+use libtest_mimic::{Failed, Trial};
+use probe_rs::debug::{DebugInfo, DebugRegisters};
+use probe_rs::flashing::{FileDownloadError, Format};
+use probe_rs::{
+    exception_handler_for_core, BreakpointCause, Core, CoreInterface, CoreStatus, Error,
+    HaltReason, Lister, MemoryInterface, SemihostingCommand, Session, VectorCatchCondition,
+};
+use probe_rs_target::MemoryRegion;
+use signal_hook::consts::signal;
+use static_cell::StaticCell;
+use time::UtcOffset;
+
+use crate::util::common_options::{BinaryDownloadOptions, ProbeOptions};
+use crate::util::flash::run_flash_download;
+use crate::util::rtt::{self, RttConfig};
+use crate::FormatOptions;
+
+const RTT_RETRIES: usize = 10;
+
+#[derive(clap::Parser)]
+pub struct Cmd {
+    #[clap(flatten)]
+    pub(crate) probe_options: ProbeOptions,
+
+    #[clap(flatten)]
+    pub(crate) download_options: BinaryDownloadOptions,
+
+    /// Always print the stacktrace on ctrl + c.
+    #[clap(long)]
+    pub(crate) always_print_stacktrace: bool,
+
+    /// Whether to erase the entire chip before downloading
+    #[clap(long)]
+    pub(crate) chip_erase: bool,
```

Comment:
I thought we have a field for this in the shared options already 🤔

Reply:
not really...